From 0d42d7e12d23d007775f4ce205506db877852eac Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Tue, 13 Oct 2020 14:34:26 +0200
Subject: [PATCH 1/2] arm64: dt: imx8mq-mba8mx: add GPIO for SD card supply

With mainboard rev.030x the SD RESET signal ca be used to switch of
the SD card supply. Athough this is more important for other SOM using
this board, the GPIO should be under control of the SD card driver to
prevent irregular behavour when using the signal as normal GPIO.

Since the signal has no meaning and is not connecte for HW rev.0200
we do not need to take special care for the older revision.

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 arch/arm64/boot/dts/freescale/imx8mq-mba8mx.dtsi | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/imx8mq-mba8mx.dtsi b/arch/arm64/boot/dts/freescale/imx8mq-mba8mx.dtsi
index 39f30499c396..316ae7e17fa8 100644
--- a/arch/arm64/boot/dts/freescale/imx8mq-mba8mx.dtsi
+++ b/arch/arm64/boot/dts/freescale/imx8mq-mba8mx.dtsi
@@ -151,6 +151,8 @@
 		regulator-name = "VSD_3V3";
 		regulator-min-microvolt = <3300000>;
 		regulator-max-microvolt = <3300000>;
+		gpio = <&gpio2 19 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
 	};
 
 	reg_12v: regulator-12v {
@@ -382,6 +384,7 @@
 	pinctrl_usdhc2_gpio: usdhc2grpgpio {
 		fsl,pins = <
 			MX8MQ_IOMUXC_SD2_CD_B_GPIO2_IO12	0x41
+			MX8MQ_IOMUXC_SD2_RESET_B_GPIO2_IO19	0xc1
 		>;
 	};
 
-- 
2.17.1

